controllers/drplacementcontrol_controller.go

```diff
@@ -1761,23 +1758,38 @@ func (r *DRPlacementControlReconciler) updateDRPCStatus(
 	return nil
 }
 
+// updateResourceCondition updates DRPC status sub-resource with updated status from VRG if one exists,
+// - The status update is NOT intended for a VRG that should be cleaned up on a peer cluster
+// It also updates DRPC ConditionProtected based on current state of VRG.
 func (r *DRPlacementControlReconciler) updateResourceCondition(
-	drpc *rmn.DRPlacementControl,
-	clusterName, vrgNamespace string,
-	log logr.Logger,
+	drpc *rmn.DRPlacementControl, userPlacement client.Object,
 ) {
-	annotations := make(map[string]string)
+	vrgNamespace, err := selectVRGNamespace(r.Client, r.Log, drpc, userPlacement)
+	if err != nil {
+		r.Log.Info("Failed to select VRG namespace", "error", err)
+
+		return
+	}
+
+	clusterName := r.clusterForVRGStatus(drpc, userPlacement, r.Log)
+	if clusterName == "" {
+		r.Log.Info("Unable to determine managed cluster from which to inspect VRG, " +
+			"skipping processing ResourceConditions")
```

Comment:
@ShyamsundarR you still need to return here right? Or handle it similar to line 1791. Lines, 1833, 1839 and 1854 can return an empty `clusterName` if there isn't a valid `clusterDecision`

The reason why this is necessary is that if the `clusterName` is empty, then `GetVRGFromManagedCluster` will look in the default namespace. An empty namespace is equivalent to the `default` namespace.

Reply:
Ack! did not notice it at all, updated now.